### PR TITLE
ENH: more dev guidelines

### PR DESCRIPTION
**Pull request by dengemann**

### CODE REVIEW

File: doc/source/contributing.rst [hunk 1]
```diff
@@ -126,6 +128,17 @@ More mne-python specific guidelines
 
     nosetests --with-timer --timer-top-n 30
 
+* Instance methods that update the state of the object should return self.
+* Use single quotes whenever possible.
+* Prefer generator or list comprehensions over ``filter``, ``map`` and other functional idioms.
+* Use explicit functional constructors for builtin containers to improve readability. E.g. ``list()``, ``dict()``.
+* Avoid nested functions if not necessary and use private functions instead.
+* When adding visualization methods, add public functions to the mne.viz package and use these in the corresponding method.
+* If not otherwise required, methods should deal with state while functions should return copies. Threre are a few justified excpetions though, e.g. ``equalize_channels``, for memory reasons for example.
```
Comment (teonlamont) [inline doc/source/contributing.rst]: Threre -> There

File: doc/source/contributing.rst [hunk 2]
```diff
@@ -126,6 +128,17 @@ More mne-python specific guidelines
...
+* Update whats_new at the end, otherwise merge conflicts are garuanteed to occur.
```
Comment (teonlamont) [inline doc/source/contributing.rst]: garuanteed -> guaranteed

**Comment by Eric89GXL** [comment]
LGTM. BTW @dengemann if you can add `[ci skip]` to commits that only affect things that don

**Comment by Eric89GXL** [comment]
't run in tests like changes to `doc/*` files, it would help prevent clogging up Travis/AppVeyor

**Comment by Eric89GXL** [comment]
(ignore CI feedback, I killed the builds because they are not affected by `doc/` changes)

**Comment by Eric89GXL** [comment]
@dengemann also please check to see if you have satisfied your requirements :)

https://github.com/mne-tools/mne-python/issues/1109#issuecomment-97202019

**Comment by dengemann** [comment]
@Eric89GXL [ci skip] to the commit message?

**Comment by Eric89GXL** [comment]
`git commit -am 'FIX: Fix typos [ci skip]'` will make it so the CIs don't run on the commit

**Comment by Eric89GXL** [comment]
Beautiful :)

**Comment by Eric89GXL** [comment]
LGTM, +1 for merge. You done @dengemann?

File: doc/source/contributing.rst [hunk 3]
```diff
@@ -126,6 +128,18 @@ More mne-python specific guidelines
...
+* Use single quotes whenever possible.
+* Prefer generator or list comprehensions over ``filter``, ``map`` and other functional idioms.
+* Use explicit functional constructors for builtin containers to improve readability. E.g. ``list()``, ``dict()``.
+* Avoid nested functions if not necessary and use private functions instead.
+* When adding visualization methods, add public functions to the mne.viz package and use these in the corresponding method.
+* If not otherwise required, methods should deal with state while functions should return copies. There are a few justified exceptions though, e.g. ``equalize_channels``, for memory reasons for example.
+* Update the whats_new.rst file at the end, otherwise merge conflicts are guaranteed to occur.
+* Avoid **kwargs and *args in function signatures, they are not user friendly (inspection).
+* Avoid single character variable names if you can. They are not readable and often they don't comply with the builtin debugger.
+* Add at least some brief comment to a private function to help us guess what it does. For complex private functions please wirte a full documentation.
```
Comment (agramfort) [inline doc/source/contributing.rst]: wirte -> write

besides +1 for merge

**Comment by agramfort** [comment]
after merging reorg branch and this one can update the github website + add
redirect so we can see how it looks like?

after we should be good to go...

**Comment by Eric89GXL** [comment]
I added the redirect, but I'm not on my computer that I was using to build the website anymore. @agramfort you want to see if you can get your build working? We want to switch to that eventually anyway.

**Comment by agramfort** [comment]
yes building works fine here

**Pull request merged**